crates/stages/src/stages/bodies.rs

```diff
@@ -87,31 +87,35 @@ impl<DB: Database, D: BodyDownloader, C: Consensus> Stage<DB> for BodyStage<D, C
         }
 
         // The block we ended at last sync, and the one we are starting on now
-        let previous_block = input.stage_progress.unwrap_or_default();
-        let starting_block = previous_block + 1;
+        let stage_progress = input.stage_progress.unwrap_or_default();
+        let starting_block = stage_progress + 1;
 
         // Short circuit in case we already reached the target block
         let target = previous_stage_progress.min(starting_block + self.commit_threshold);
-        if target <= previous_block {
-            return Ok(ExecOutput { stage_progress: target, reached_tip: true, done: true })
+        if target <= stage_progress {
+            return Ok(ExecOutput { stage_progress, reached_tip: true, done: true })
         }
 
         let bodies_to_download = self.bodies_to_download::<DB>(db, starting_block, target)?;
 
-        // Cursors used to write bodies and transactions
+        // Cursors used to write bodies, ommers and transactions
+        let mut body_cursor = db.cursor_mut::<tables::BlockBodies>()?;
         let mut ommers_cursor = db.cursor_mut::<tables::BlockOmmers>()?;
         let mut tx_cursor = db.cursor_mut::<tables::Transactions>()?;
-        let mut tx_count_cursor = db.cursor_mut::<tables::CumulativeTxCount>()?;
+
+        // Cursors used to write state transition mapping
+        let mut block_transition_cursor = db.cursor_mut::<tables::BlockTransitionIndex>()?;
+        let mut tx_transition_cursor = db.cursor_mut::<tables::TxTransitionIndex>()?;
 
         // Get id for the first transaction in the block
-        let mut first_tx_id = db.get_first_tx_id(starting_block)?;
+        let (mut current_tx_id, mut transition_id) = db.get_next_block_ids(starting_block)?;
```

Comment:
Gets the next transaction id we can write to for transactions, and the block transition id for the block reward